apps/api/src/server/middleware/auth-handler.js

```diff
@@ -0,0 +1,72 @@
+import config from '@pins/web/environment/config.js';
+import jwt from 'jsonwebtoken';
+import jwksClient from 'jwks-rsa';
+import logger from '../utils/logger.js';
+
+const DISCOVERY_KEYS_ENDPOINT = `${config.msal.authority}/discovery/v2.0/keys`;
+
+const client = jwksClient({
+	cache: true,
+	rateLimit: true,
+	jwksUri: DISCOVERY_KEYS_ENDPOINT
+});
+
+const jwtOptions = {
+	audience: [`api://${config.msal.clientId}`],
+	issuer: [`https://sts.windows.net/${process.env['AUTH_TENANT_ID']}/`],
+	algorithms: ['RS256']
+};
+
+/**
+ * The auth handler.
+ *
+ * @type {import('express').Handler}
+ */
+export async function validateClientHandler(req, response, next) {
+	// 1) Getting token and check of it's there
+	let accessToken;
+
+	const errorMessage = 'You have no access! Please try again later.';
+
+	try {
+		if (req.headers.authorization && req.headers.authorization.startsWith('Bearer')) {
+			accessToken = req.headers.authorization.split(' ')[1];
+		}
+
+		if (!accessToken) {
+			throw new Error(errorMessage);
+		}
+
+		logger.info('attempting to decode access token');
+
+		const decoded = jwt.decode(accessToken, { complete: true });
```

Comment:
This is not a requirement, but a suggestion: I would move these single lines of code to their own functions, such as 
```
const getDecodedAccessToken = (accessToken) => {
    return jwt.decode(accessToken, { complete: true })
}
```
the reason I am suggesting this is because this function has many responsibilities and there is difficult to digest when reading it

Reply:
will do that makes sense